profiling/src/profiling.rs

```diff
@@ -739,199 +739,179 @@ impl Profiler {
-                        Ok(tag) => tags.push(tag),
-                        Err(err) => warn!("Tag error: {}", err),
-                    }
-                }
-
-                let message = SampleMessage {
-                    key: ProfileIndex {
-                        sample_types,
-                        tags,
-                        endpoint: locals.uri.clone(),
-                    },
-                    value: SampleData {
-                        frames,
-                        labels: labels.clone(),
-                        sample_values,
-                    },
-                };
-
-                match self.send_sample(message) {
+                let labels = self.message_labels();
+
+                match self.send_sample(self.prepare_sample_message(
+                    frames,
+                    0,
+                    0,
+                    None,
+                    count,
+                    bytes,
+                    &labels,
+                    locals
+                )) {
                     Ok(_) => trace!(
-                        "Sent stack sample of depth {} with labels {:?} to profiler.",
-                        depth,
-                        labels
+                        "Sent stack sample of depth {depth} with size {bytes}, labels {labels:?} and count {count} to profiler."
                     ),
                     Err(err) => warn!(
-                        "Failed to send stack sample of depth {} with labels {:?} to profiler: {}",
-                        depth, labels, err
+                        "Failed to send stack sample of depth {depth} with size {bytes}, labels {labels:?} and count {count} to profiler: {err}"
                     ),
                 }
             }
             Err(err) => {
-                warn!("Failed to collect stack sample: {}", err)
+                warn!("Failed to collect stack sample: {err}")
+            }
+        }
+    }
+
+    fn message_labels(&self) -> Vec<Label> {
+        let mut labels = vec![];
+        let gpc = unsafe { datadog_php_profiling_get_profiling_context };
+        if let Some(get_profiling_context) = gpc {
+            let context = unsafe { get_profiling_context() };
+            if context.local_root_span_id != 0 {
+                labels.push(Label {
+                    key: "local root span id".into(),
+                    value: LabelValue::Str(format!("{}", context.local_root_span_id).into()),
+                });
+                labels.push(Label {
+                    key: "span id".into(),
+                    value: LabelValue::Str(format!("{}", context.span_id).into()),
+                });
+            }
+        }
+        labels
+    }
+
+    fn prepare_sample_message(
+        &self,
+        frames: Vec<ZendFrame>,
+        sample: i64,
+        wall_time: i64,
+        cpu_time: Option<i64>,
+        alloc_samples: i64,
+        alloc_size: i64,
+        labels: &Vec<Label>,
```

Comment:
Since this is read-only, clippy suggests `&[Label]` instead:

```
warning: writing `&Vec` instead of `&[_]` involves a new object where a slice will do
   --> src/profiling.rs:855:17
    |
855 |         labels: &Vec<Label>,
    |                 ^^^^^^^^^^^
    |
    = note: `#[warn(clippy::ptr_arg)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg
help: change this to
    |
855 ~         labels: &[Label],
856 |         locals: &RequestLocals,
857 |     ) -> SampleMessage {
858 |         let mut sample_types = vec![
859 |             ValueType {
860 |                 r#type: Cow::Borrowed("sample"),
```